File: model/src/main/java/com/grocerylist/model/Dish.java
package com.grocerylist.model;

import lombok.Data;

import javax.persistence.*;
import java.util.List;
/**
 * @author  Adam Mendak
 * @version 1.0
 * @since   2019-10-19
 *
 * Abstract class which contains fields for holding entered Dish data
 */
@Data
@Entity
@Table(name = "DISH")
public class Dish extends AbstractTimestampableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "DESCRIPTION")
    private String description;

    @Column(name = "PREPARE_TIME")
    private Integer prepareTime;

    @Column(name = "DIFFICULTY_LEVEL")
    private String difficultyLevel;

    @Column(name = "NUMBER_OF_SERVINGS")
    private Integer numberOfServings;

    @ManyToMany
    @JoinTable(name = "DISH_DISH_CATEGORY",
            joinColumns = @JoinColumn(name = "dish_id"),
            inverseJoinColumns = @JoinColumn(name = "dish_category_id")
    )
    private List<DishCategory> categories;

    @ManyToMany(cascade = CascadeType.REMOVE)
    @JoinTable(name = "DISH_INGREDIENT",
            joinColumns = @JoinColumn(name = "dish_id"),
            inverseJoinColumns = @JoinColumn(name = "ingredient_id")
    )
    private List<Ingredient> ingredients;

}

File: frontend/groceryFront/src/app/dish/dish.ts
export interface IDish {
  id: number;
  description: string;
  prepareTime: number;
  difficultyLevel: string;
  numberOfServings: number;
  photoURL: string;
}

export class Dish implements IDish {
  private _id: number;
  private _description: string;
  private _prepareTime: number;
  private _difficultyLevel: string;
  private _numberOfServings: number;
  private _photoURL: string;

  setId(id: number): void {
    this._id = id;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get prepareTime(): number {
    return this._prepareTime;
  }

  set prepareTime(value: number) {
    this._prepareTime = value;
  }

  get difficultyLevel(): string {
    return this._difficultyLevel;
  }

  set difficultyLevel(value: string) {
    this._difficultyLevel = value;
  }

  get numberOfServings(): number {
    return this._numberOfServings;
  }

  set numberOfServings(value: number) {
    this._numberOfServings = value;
  }

  get photoURL(): string {
    return this._photoURL;
  }

  set photoURL(value: string) {
    this._photoURL = value;
  }
}
